tests/boot/boot_to_desktop.pm

```diff
@@ -42,4 +42,29 @@ sub test_flags {
     return {fatal => 1, milestone => 1};
 }
 
+sub post_fail_hook {
+    my $self = shift;
+
+    save_screenshot;
+
+    # check for text login to check if X has failed
+    if (check_screen('generic-login')) {
+        record_info 'Seems that the display manager failed';
+    }
+
+    # if we found a shell, we do not need the memory dump
+    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
+        die "save_memory_dump not implemented, no way to save memory_dump" unless check_var('BACKEND', 'qemu');
+        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
+        save_memory_dump;
+    }
+
+    # crosscheck for text login on tty1
+    select_console 'root-console';
+
+    # collect and upload some stuff
+    $self->export_logs();
+
+}
```

Comment:
So you are overwriting the base class post_fail_hook? It is still useful, e.g. the detection for blocked kernel tasks

Reply:
that's addressed with `SUPER::post_fail_hook();` no?

Reply:
yes, looks good now.